tests/tape/test_qscript.py

```diff
@@ -45,6 +45,7 @@ def test_no_update_empty_initialization(self):
         assert len(qs._trainable_params) == 0
         assert qs._graph is None
         assert qs._specs is None
+        assert qs._shots.total_shots is None
```

Comment:
Made a story to add `__eq__` and hash comparison to the `Shots` class. Then we can do `Shots(None) == Shots(None)`.   But for now, is good.